nixos/lib/test-driver/src/test_driver/__init__.py

```diff
@@ -142,12 +143,22 @@ def main() -> None:
             history_dir = os.getcwd()
             history_path = os.path.join(history_dir, ".nixos-test-history")
             if offset := args.dump_vsocks:
+                if not Path("/dev/vhost-vsock").is_char_device():
+                    logger.error(
+                        "`/dev/vhost-vsock` doesn't exist, but the vsocks backdoor is enabled. Is this feature not supported on your system?"
+                    )
+                    sys.exit(1)
                 driver.dump_machine_ssh(offset)
             ptpython.ipython.embed(
                 user_ns=driver.test_symbols(),
                 history_filename=history_path,
             )  # type:ignore
         else:
+            if args.dump_vsocks and not Path("/dev/vhost-vsock").is_char_device():
+                logger.error(
+                    "`/dev/vhost-vsock` doesn't exist, but vsock backdoor is enabled for a non-interactive run. Are you trying to run a VM test in the sandbox with `sshBackdoor.enable = true;`?"
+                )
```

Comment:
looking at https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/testing/interactive.nix, i think we could simply add an assertion that fires if the backdoor was enabled in the non-interactive config part and warn/fail about it.

this would be good because:

- it provides actionable feedback on what is wrong and how to fix it
- we can remove run time code from the python driver as the case we need to defend against is illegal to express.